ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/search-settings/search-settings.tsx

```diff
@@ -0,0 +1,176 @@
+  .editor-footer {
+    display: none;
+  }
+  .editor-properties {
+    padding: 0px;
+  }
+`
+
+const EditorFooter = styled<Props, 'div'>('div')`
+  display: ${props => (props.showFooter ? 'block' : 'none')};
+  button {
+    display: inline-block;
+    width: 50%;
+  }
+`
+
+type Props = {
+  onClose?: () => void
+  model?: any
+  showFooter: boolean
+} & WithBackboneProps
+
+export default hot(module)(
+  withListenTo(
+    class SearchSettings extends React.Component<Props, {}> {
+      propertyModel: any
+      queryModel: any
+      querySettingsView: any
+      propertyView: any
+      constructor(props: Props) {
+        super(props)
+        this.setQuerySettingsView()
+        this.setPropertyView()
+      }
+      render() {
+        return (
+          <Root>
+            <div className="editor-properties">
+              <PropertyResultCount>
+                <MarionetteRegionContainer
+                  view={this.propertyView}
+                  replaceElement
+                />
+              </PropertyResultCount>
+              <div className="is-header">Defaults</div>
+              <QuerySettings className="property-search-settings">
+                <MarionetteRegionContainer view={this.querySettingsView} />
+              </QuerySettings>
+            </div>
+            <EditorFooter {...this.props}>
+              <button className="is-negative" onClick={this.triggerCancel}>
+                <span className="fa fa-times" />
+                <span>Cancel</span>
+              </button>
+              <button className="is-positive" onClick={this.triggerSave}>
+                <span className="fa fa-floppy-o" />
+                <span>Save</span>
+              </button>
+            </EditorFooter>
+          </Root>
+        )
+      }
+      triggerSave = () => {
+        this.updateResultCountSettings()
+        this.updateSearchSettings()
+        this.props.listenTo(
+          ConfirmationView.generateConfirmation({
+            prompt: 'Do you want to apply the new defaults to this search?',
+            no: 'No',
+            yes: 'Apply',
+          }),
+          'change:choice',
+          (confirmation: any) => {
+            if (confirmation.get('choice')) {
+              this.props.model.applyDefaults()
+            }
+          }
+        )
+        user.savePreferences()
+        if (!!this.props.onClose) {
+          this.props.onClose()
+        }
+      }
+      updateResultCountSettings = () => {
+        user.getPreferences().set({
+          resultCount: this.propertyModel.getValue()[0],
+        })
+      }
+      updateSearchSettings = () => {
+        user
+          .getPreferences()
+          .get('querySettings')
+          .set(this.querySettingsView.toJSON())
+      }
+      triggerCancel = () => {
+        this.setPropertyView()
+        this.setQuerySettingsView()
+        if (!!this.props.onClose) {
+          this.props.onClose()
+        }
+        this.forceUpdate()
```

Comment:
❓ Why is this necessary here?

Reply:
I think it's used to re-render the QuerySettingsView. We should be keeping that view in the state since it directly affects rendering

Reply:
i tried putting the views in the state, and having a setState that re-instantiates those view objects but it still doesn't reset the value.